homeassistant/components/rest/sensor.py

```diff
@@ -99,35 +128,41 @@ def __init__(
         coordinator: DataUpdateCoordinator[None] | None,
         rest: RestData,
         config: ConfigType,
-        unique_id: str | None,
+        trigger_entity_config: ConfigType,
     ) -> None:
         """Initialize the REST sensor."""
+        ManualTriggerSensorEntity.__init__(self, hass, trigger_entity_config)
         RestEntity.__init__(
             self,
             coordinator,
             rest,
             config.get(CONF_RESOURCE_TEMPLATE),
             config[CONF_FORCE_UPDATE],
         )
-        TemplateSensor.__init__(
-            self,
-            hass,
-            config=config,
-            fallback_name=DEFAULT_SENSOR_NAME,
-            unique_id=unique_id,
-        )
         self._value_template = config.get(CONF_VALUE_TEMPLATE)
         if (value_template := self._value_template) is not None:
             value_template.hass = hass
         self._json_attrs = config.get(CONF_JSON_ATTRS)
         self._json_attrs_path = config.get(CONF_JSON_ATTRS_PATH)
+        self._attr_extra_state_attributes = {}
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        available1 = RestEntity.available.fget(self)  # type: ignore[attr-defined]
+        available2 = ManualTriggerSensorEntity.available.fget(self)  # type: ignore[attr-defined]
+        return bool(available1 and available2)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return extra attributes."""
+        return dict(self._attr_extra_state_attributes)
 
     def _update_from_rest_data(self) -> None:
         """Update state from the rest data."""
         value = self.rest.data_without_xml()
 
         if self._json_attrs:
-            self._attr_extra_state_attributes = {}
```

Comment:
Note: this changes the behavior in case of attribute parsing failure.
- previously, a failure would cause the attributes to a blank dict
- now, a failure causes the previous attributes to be kept

What this change on purpose? (also see #97526)

Reply:
We should not change this behavior in this PR at least.
So I see you have fixed that in the other PR so it comes back.
Thanks 👍